### DISCUSSION TITLE
Custom Size behavior

### DISCUSSION THREAD

**Discussion by robdejonge** [rxhanson/RectanglePro-Community#364] [Ideas]
So I like 1200x900 as a window size for some apps, such as Mail. To this effect, I have set up a preset in the Custom Size & Position list. I have 1 suggestion and 1 question about what I'm seeing next. 

Suggestion: Like the 'Display' setting, implement an 'Unchanged' Docking Position where I don't see the X and Y fields, and have to leave them empty. 

Question: When I hit the keyboard shortcut, the window does indeed resize to the configured W and H. But as soon as I then drag the window away from the location it is in, it pops back to the size it had prior to the resize. Why is that?! 

But really liking the look of this!

**Comment by rxhanson** [comment]
Thanks for the feedback!

> Suggestion: Like the 'Display' setting, implement an 'Unchanged' Docking Position where I don't see the X and Y fields, and have to leave them empty.

I think what you're looking for can be achieved by selecting "custom" and just leaving the fields empty. I see what you mean, though, and will think about this for a future release.

> Question: When I hit the keyboard shortcut, the window does indeed resize to the configured W and H. But as soon as I then drag the window away from the location it is in, it pops back to the size it had prior to the resize. Why is that?!

Sort of unintuitive, but there's a checkbox in the Snap Areas tab of the prefs window for "Restore window size when unsnapped". If you uncheck that box then the window won't do that.

**Comment by robdejonge** [comment]
Love it, thanks!

Continuing the migration of my home-built window manager to using Rectangle Pro. Mine was built with a gap around the edge of main windows too, with the specific reason for that being so underlying windows are just a click away. Rectangle Pro implementing this gap setting too is great. 

However, I have a few chat windows that I set up to be right up against the screen edge. This way they peak out from under the browser windows and, as intended, are just a click away. Using the 'custom' feature, I can position and size the window correctly, but the 'gap' setting (I assume) then kicks in and moves the window back 'under' my browser window where I spend most of my time. 

Stash mode is a great idea, but does not work for what I am trying to achieve here. At least not in my particular setup.

Suggestion: 
Add a toggle in every 'custom setting' to override the gap. If enabled, the aforementioned correction does not happen. 

Please don't interpret my suggestions as criticism. I've already been raving about Rectangle Pro to several people and I will for sure buy a license. It's passion, is what it is! :-) 

(also: do you use these issues as your feature request backlog, or do I close an issue once you've answered my questions?)

**Comment by robdejonge** [comment]
Purchased my license! Love it!

**Comment by rxhanson** [comment]
Thanks! I converted this over to a discussion so I can still keep track of your suggestions here.
